src/helpers.rs

```diff
@@ -38,12 +37,24 @@ pub fn validate_native_denom(denom: &str) -> Result<(), ContractError> {
 
 // Validates that the channel ID is of the form `channel-N`
 pub fn validate_channel_id(channel_id: &str) -> Result<(), ContractError> {
-    let re = Regex::new(CHANNEL_REGEX).unwrap();
-    if !re.is_match(channel_id) {
+    let Some((prefix, id)) = channel_id.split_once('-') else {
```

Comment:
I think we can consolidate this with

```rust
const CHANNEL_PREFIX: &str = "channel-";

    match channel_id.strip_prefix(CHANNEL_PREFIX) {
        Some(id) if id.parse::<u64>().is_ok() => Ok(()),
        _ => return Err(ContractError::InvalidChannelID {
            channel_id: channel_id.to_string(),
        })
    }
```

But it might also be a little less readable, wdyt?

Reply:
I leave it with what you have since you're the rust 👑